From 98561cac65445feb44b3d615c585b41e8bdb337c Mon Sep 17 00:00:00 2001
From: Connor Rigby <connorr@hey.com>
Date: Sun, 31 Jul 2022 15:14:54 -0600
Subject: [PATCH] Fix build for esp32s3

---
 CMakeLists.txt | 23 +++++++++++++++++++----
 config_ext.h   |  2 +-
 esp32.c        | 11 ++++++++++-
 3 files changed, 30 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index afee6e1..88ed302 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,9 +1,24 @@
 set(COMPONENT_ADD_INCLUDEDIRS "include")
 
 idf_component_register(SRCS "sqlite3.c" "esp32.c" "shox96_0_2.c"
-                       INCLUDE_DIRS "include"
-                       PRIV_INCLUDE_DIRS "private_include"
-                       REQUIRES mbedtls
-                       PRIV_REQUIRES console spiffs)
+INCLUDE_DIRS "include"
+PRIV_INCLUDE_DIRS "private_include"
+REQUIRES mbedtls
+PRIV_REQUIRES console spiffs)
+set_source_files_properties(
+sqlite3.c
+PROPERTIES COMPILE_FLAGS
+"-Wno-unused-function -Wno-unused-variable -Wno-unused-value -Wno-unused-label -Wno-missing-field-initializers -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-unused-parameter -Wno-implicit-fallthrough -Wno-cast-function-type"
+)
+set_source_files_properties(
+esp32.c
+PROPERTIES COMPILE_FLAGS
+"-Wno-unused-function -Wno-unused-variable -Wno-unused-value -Wno-extra -Wno-type-limits -Wno-unused-label -Wno-missing-field-initializers -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-unused-parameter -Wno-implicit-fallthrough -Wno-cast-function-type"
+)
+set_source_files_properties(
+shox96_0_2.c
+PROPERTIES COMPILE_FLAGS
+"-Wno-unused-function -Wno-unused-variable -Wno-char-subscripts -Wno-unused-value -Wno-unused-label -Wno-missing-field-initializers -Wno-unused-but-set-variable -Wno-maybe-uninitialized -Wno-unused-parameter -Wno-implicit-fallthrough -Wno-cast-function-type"
+)
 
 target_compile_options(${COMPONENT_LIB} PRIVATE -std=gnu99 -g3 -fno-stack-protector -ffunction-sections -fdata-sections -fstrict-volatile-bitfields -mlongcalls -nostdlib -Wpointer-arith -Wno-error=unused-value -Wno-error=unused-label -Wno-error=unused-function -Wno-error=unused-but-set-variable -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wno-error=char-subscripts -Wno-error=maybe-uninitialized -Wno-unused-parameter -Wno-sign-compare -Wno-old-style-declaration -MMD -c -DF_CPU=240000000L -DESP32 -DCORE_DEBUG_LEVEL=0 -DNDEBUG)
diff --git a/config_ext.h b/config_ext.h
index 0e3035a..09bc554 100644
--- a/config_ext.h
+++ b/config_ext.h
@@ -1,7 +1,7 @@
 #define BUILD_sqlite -DNDEBUG
 #define SQLITE_CORE                          1
 #define SQLITE_NO_SYNC                       1
-#define YYSTACKDEPTH                        20
+#define YYSTACKDEPTH                         0
 #define SQLITE_TEMP_STORE                    1
 #define SQLITE_SYSTEM_MALLOC                 1
 #define SQLITE_OS_OTHER                      1
diff --git a/esp32.c b/esp32.c
index f6be8a2..88d5948 100644
--- a/esp32.c
+++ b/esp32.c
@@ -16,7 +16,16 @@
 #include <sqlite3.h>
 #include <esp_spi_flash.h>
 #include <esp_system.h>
-#include <rom/ets_sys.h>
+#include <esp_random.h>
+
+#if CONFIG_IDF_TARGET_ESP32
+#include "esp32/rom/ets_sys.h"
+#elif CONFIG_IDF_TARGET_ESP32S2
+#include "esp32s2/rom/ets_sys.h"
+#elif CONFIG_IDF_TARGET_ESP32S3
+#include "esp32s3/rom/ets_sys.h"
+#endif
+
 #include <sys/stat.h>
 
 #include "shox96_0_2.h"
-- 
2.35.1

